src/fsharp/TastPickle.fs

```diff
@@ -548,8 +560,22 @@ let u_array_ext extraf f st =
     let arr = u_array_core f (n &&& 0x7FFFFFFF) st
     extraItem, arr
 
-let u_list f st = Array.toList (u_array f st)
-let u_list_ext extra f st = let v, res = u_array_ext extra f st in v, Array.toList res
+let u_list_core f n st =
+    [ for _ in 1..n do
```

Comment:
This allocates more than the original code.  A list generation expression has an implicit allocation of a `ResizeArray` followed by conversion to an array.

It's no big deal but just letting you know - the original code is better I think

Reply:
I'll try to test it

Reply:
@dsyme for the code:
```
let ls =
    [ for i in 1..10 do
        yield i ]
```

this is the allocations I get from the visual studio memory profiler
![image](https://user-images.githubusercontent.com/1097246/49592088-e3eaae80-f978-11e8-9d75-445cdb6ed099.png)

Reply:
when trying the array generation expression i get this ResizeArray allocation

Reply:
Sheesh that should be improved.